### PR TITLE
tests/pkg_semtech_loramac: improve README

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR improves the README of semtech loramac test.
* Add OTAA and ABP sections in `Using the shell` section
* Add RX2 datarate for receiving downlink and confirmed messages when using ABP mode and The Things Network backend.
* Add `sending and receiving data` and `other shell commands` sections

### Issues/PRs references
None so far

### CODE REVIEW

File: tests/pkg_semtech-loramac/README.md [hunk 1]
```diff
@@ -147,4 +147,11 @@ The node will also print the data received:
 ```
     > loramac tx test
     Data received: This is RIOT!
-```
\ No newline at end of file
+```
+
+### The Things Network
+
+If using TTN make sure to set the correct datarate for RX2. Otherwise confirmed and downlink messages won't work. The datarate for RX2 should be DR3 (SF9BW125) as seen in [TTN LoRaWAN Frequencies Overview](https://www.thethingsnetwork.org/docs/lorawan/frequency-plans.html):
```
Comment (aabadie) [inline tests/pkg_semtech-loramac/README.md]: Can you split this line to have it within 80 characters length ?

**Comment by jia200x** [comment]
@aabadie done!

File: tests/pkg_semtech-loramac/README.md [hunk 2]
```diff
@@ -147,4 +147,14 @@ The node will also print the data received:
 ```
     > loramac tx test
     Data received: This is RIOT!
-```
\ No newline at end of file
+```
+
+### The Things Network
```
Comment (aabadie) [inline tests/pkg_semtech-loramac/README.md]: In fact, I would move this before the MQTT related stuff. Maybe the README should be reorganized as follows:

Keep section "Using the shell" but refactor it slightly with:
- a subsection "OTAA activation"
- a subsection "ABP activation", where information is given with required keys/EUI and the trick with the RX2 setting
- a subsection with sending receiving messages
- a subsection "Other shell commands" with other setting commands (DR, ADR, etc) plus the shell help

What do you think @jia200x ?

Comment (jia200x) [reply tests/pkg_semtech-loramac/README.md]: yes, I agree. I can use this same PR

**Comment by jia200x** [comment]
@aabadie I also added information about the default class, and how data is received

**Comment by jia200x** [comment]
>I had to reset the frame counter though. Maybe this could be mentioned in the README ?

That's because of the LoRaWAN specification. In theory, frame counters should be stored in FLASH. Shall we add it anyway?

**Comment by aabadie** [comment]
> In theory, frame counters should be stored in FLASH. Shall we add it anyway?

I know but this is a limitation of the actual loramac support in RIOT. It could be added using the eeprom periph API (only on L0/L1) or with the flashpage periph API.
I think we should add it for ABP activation because otherwise it looks like it doesn't work.

**Comment by jia200x** [comment]
@aabadie addressed!

**Comment by aabadie** [comment]
Ah, the CI reports white space errors, can you fix them ?

**Comment by jia200x** [comment]
fixed! let's wait for Murdock

**Comment by aabadie** [comment]
Thanks @jia200x, let's merge now !

**Pull request merged**